src/decode.rs

```diff
@@ -708,7 +708,8 @@ fn order_palette(
 unsafe fn read_pal_indices(
     ts: &mut Rav1dTileState,
     scratch_pal: &mut Rav1dTaskContext_scratch_pal,
-    pal_idx: &mut [u8],
+    pal_tmp: &mut [u8],
```

Comment:
Would it make sense to pass a reference to `t.scratch` instead? @randomPoison has been working on this structure and might have an opinion.